### PR TITLE
[#6256,#7220,#7519,#7545,#7546,irods/irods_resource_plugin_s3#2156] update externals (4-3-stable)

### PR DESCRIPTION
**Pull request by SwooshyCueb**
Merge irods/externals#212 and irods/irods_development_environment#128, then update externals package repos before merging
In service of #6256
In service of #7220
In service of #7519
In service of #7545
In service of #7546
In service of irods/irods_resource_plugin_s3#2156

Companion PRs:
- irods/irods_rule_engine_plugin_audit_amqp#132
- irods/irods_api_plugin_genquery2#90
- irods/irods_capability_storage_tiering#261
- irods/irods_client_http_api#257
- irods/irods_microservice_plugins_curl#64
- irods/irods_netcdf#52
- irods/irods_resource_plugin_s3#2173
- irods/irods_rule_engine_plugin_logical_quotas#106
- irods/irods_rule_engine_plugin_metadata_guard#75
- irods/irods_rule_engine_plugin_python#193

### CODE REVIEW

**Comment by SwooshyCueb** [comment]
Marked ready for review

**Comment by trel** [comment]
seems good to my eye.   value accordingly :)

**Review by korydraughn — DISMISSED** [review]
Looks good to me.

Pound it.

**Comment by SwooshyCueb** [comment]
There was a suggested change in the main branch equivalent of this PR (#7526), so I'm pulling this back into draft mode to bring it to parity.

**Comment by SwooshyCueb** [comment]
Marked ready for review

**Review by korydraughn — APPROVED** [review]
Looks good to me.

Let's get one more approval before squashing/pounding.

File: CMakeLists.txt [hunk 1]
```diff
@@ -92,6 +72,37 @@ project(irods
...
+# This is duplicated from UseLibCXX.cmake, since we need it before UseLibCXX can be included.
+if (CMAKE_CXX_COMPILER_ID MATCHES ".*Clang")
+  set(IRODS_BUILD_AGAINST_LIBCXX_DEFAULT ON)
+else()
+  set(IRODS_BUILD_AGAINST_LIBCXX_DEFAULT OFF)
+endif()
+set(IRODS_BUILD_AGAINST_LIBCXX ${IRODS_BUILD_AGAINST_LIBCXX_DEFAULT} CACHE BOOL "Try to build against libc++ instead of libstdc++.")
+
+# NOTE: At present, our split externals packages have parity. Should this ever change, support for automatically
+# switching exernals packages based on IRODS_BUILD_AGAINST_LIBCXX should be removed.
```
[the remark this answers is not in the document]
Comment (SwooshyCueb) [reply CMakeLists.txt]: Fixed

**Comment by trel** [comment]
good here.

**Review by korydraughn — APPROVED** [review]
Pound it.

**Comment by SwooshyCueb** [comment]
\#'d

**Review by alanking — APPROVED** [review]

**Pull request merged**